src/ResolvedApi.ts

```diff
@@ -231,26 +224,7 @@ export default class ResolvedApi implements Client {
   }
 
   getPreviewResolver(token: string, documentId?: string): PreviewResolver {
-    
-    const resolve = (linkResolver: LinkResolver, defaultUrl: string, cb?: RequestCallback<string>) => {
-      if (documentId) {
-        return this.getByID(documentId, { ref: token }).then((document: Document) => {
-          if (!document) {
-            cb && cb(null, defaultUrl);
-            return defaultUrl;
-          } else {
-            const url = linkResolver(document);
-            cb && cb(null, url);
-            return url;
-          }
-        });
-      } else {
-        return Promise.resolve(defaultUrl);
-      }
-    }
-    
-   
-    return { token, documentId, resolve }
+    return createPreviewResolver(token, documentId, this.getByID.bind(this));
```

Comment:
would it be possible to have something like:
getPreviewResolver( a, b, c) = createPreviewResolver( a, b )

Or would the this not be available in this situation ?